pg_search/src/postgres/customscan/pdbscan/exec_methods/fast_fields/mod.rs

```diff
@@ -146,46 +142,118 @@ unsafe fn ff_to_datum(
 }
 
 /// Count how many "fast fields" are requested to be used by the query, as described by the `builder` argument.
-pub unsafe fn count(
+pub unsafe fn count_fast_fields(
     builder: &mut CustomPathBuilder<PrivateData>,
     rti: pg_sys::Index,
     table: &PgRelation,
     schema: &SearchIndexSchema,
     target_list: *mut pg_sys::List,
+    referenced_columns: &HashSet<pg_sys::AttrNumber>,
 ) -> f64 {
-    let ff = pullup_fast_fields(target_list, schema, table, rti).unwrap_or_default();
+    // Use our new collect function to get the fast fields
+    let ff =
+        pullup_fast_fields(target_list, referenced_columns, schema, table, rti).unwrap_or_default();
 
     builder.custom_private().set_maybe_ff(!ff.is_empty());
     ff.iter().sorted().dedup().count() as f64
 }
 
-/// Find all the fields that can be as "fast fields", categorize them as [`WhichFastField`]s, and
-/// return the list.  If there are none, or one or more of the fields can't be used as a "fast field",
-/// we return [`None`].
-pub unsafe fn collect(
+/// Find all the fields that can be used as "fast fields", categorize them as [`WhichFastField`]s,
+/// and return the list. If there are none, or one or more of the fields can't be used as a
+/// "fast field", we return [`None`].
+///
+/// This version considers all referenced columns across the query, not just those in the target list.
+pub unsafe fn collect_fast_fields(
     maybe_ff: bool,
     target_list: *mut pg_sys::List,
+    referenced_columns: &HashSet<pg_sys::AttrNumber>,
     rti: pg_sys::Index,
     schema: &SearchIndexSchema,
     heaprel: &PgRelation,
 ) -> Option<Vec<WhichFastField>> {
     if maybe_ff {
-        pullup_fast_fields(target_list, schema, heaprel, rti)
+        pullup_fast_fields(target_list, referenced_columns, schema, heaprel, rti)
     } else {
         None
     }
 }
 
+// Helper function to process an attribute number and add a fast field if appropriate
```

Comment:
`process` is a fairly generic name... since this is a helper function for `collect_fast_fields`, maybe `collect_fast_field_try_for_attno` or something...?

Reply:
Makes sense. Renamed as suggested.